### PR TITLE
feat: assessor calculates interest delta for senior tranche

### PR DESCRIPTION
**Pull request by manuelpolzhofer**

### CODE REVIEW

File: src/core/lender/assessor/assessor.sol [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+pragma solidity >=0.5.12;
+
+import "./base.sol";
+
+contract Assessor is BaseAssessor {
+
+    // accrueTrancheInterest can implement different interest models
+    function accrueTrancheInterest(address tranche_) public returns (uint) {
+        SeniorTrancheLike tranche = SeniorTrancheLike(tranche_);
+        uint debt = tranche.debt();
```
Comment (ilinzweilin) [inline src/core/lender/assessor/assessor.sol]: `assessor.accrueTrancheInterest` calls`tranche.debt()` which calls `drip()` which calls `assessor.accrueTrancheInterest()` 🙈

Comment (manuelpolzhofer) [reply src/core/lender/assessor/assessor.sol]: True. 👍  It should be `uint debt = `tranche.borrowed()+tranche.interest()`.

File: src/core/lender/assessor/default.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+contract DefaultAssessor is BaseAssessor, Interest {
+
+    // accrueTrancheInterest can implement different interest models
+    function accrueTrancheInterest(address tranche_) public returns (uint) {
+        SeniorTrancheLike tranche = SeniorTrancheLike(tranche_);
+        uint debt = tranche.borrowed() + tranche.interest();
+        // move to tinlake-math
+        // interest is calculated based on tranche debt
+        return safeSub(rmul(rpow(tranche.ratePerSecond(), now - tranche.lastUpdated(), ONE), debt), debt);
```
Comment (charleenfei) [inline src/core/lender/assessor/default.sol]: if we update the naming in tinlake-math, we can directly use `safeSub(updateChi(ONE, ratePerSecond, lastUpdated), debt)`

Comment (manuelpolzhofer) [reply src/core/lender/assessor/default.sol]: yes. we need to make the method more general.

It is not updateChi anymore.

Comment (charleenfei) [reply src/core/lender/assessor/default.sol]: yes, exactly

**Pull request merged**